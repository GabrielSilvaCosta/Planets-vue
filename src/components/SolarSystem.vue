<template>
    <div>
      <div data-testid="solar-system"></div>
      <Title headline="Planetas" />
      <PlanetCard
        v-for="planet in planets"
        :key="planet.name"
        :planetName="planet.name"
        :planetImage="planet.image"
      />
    </div>
  </template>
  
  <script>
  import Title from './Title.vue';
  import PlanetCard from './PlanetCard.vue';
  import planets from '../data/planets';
  
  export default {
    name: "SolarSystem",
    components: {
      Title,
      PlanetCard,
    },
    data() {
      return {
        planets: planets,
      };
    },
  };
  </script>

  <style scoped>

  </style>