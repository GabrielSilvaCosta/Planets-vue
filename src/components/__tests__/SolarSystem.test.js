import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import SolarSystem from "../SolarSystem.vue";

describe("SolarSystem", () => {
    it("should render correctly", () => {
        const wrapper = mount(SolarSystem, {
            props: {
                planets: [],
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
})