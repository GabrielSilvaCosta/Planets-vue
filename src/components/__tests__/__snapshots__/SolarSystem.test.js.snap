// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SolarSystem > should render correctly 1`] = `
"<div planets="">
  <div data-testid="solar-system"></div>
  <div data-v-6db1a9b7="" class="title">
    <h1 data-v-6db1a9b7="">Planetas</h1>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Mercurio</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Mercurio.png" alt="Mercurio" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Venus</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Venus.png" alt="Venus" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Terra</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Terra.png" alt="Terra" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Marte</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Marte.png" alt="Marte" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Jupiter</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Jupiter.png" alt="Jupiter" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Saturno</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Saturno.png" alt="Saturno" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Urano</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Urano.png" alt="Urano" class="planet-image"></div>
  </div>
  <div data-v-1ed930a5="" class="planet-card">
    <h3 data-v-1ed930a5="">Netuno</h3>
    <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" src="/src/images/Netuno.png" alt="Netuno" class="planet-image"></div>
  </div>
</div>"
`;
